tony-history-server/app/utils/ParserUtils.java

```diff
@@ -94,10 +94,31 @@ public static JobMetadata parseMetadata(FileSystem fs, Path jobFolderPath, Strin
     return JobMetadata.newInstance(histFileName);
   }
 
-  public static List<JobConfig> parseConfig(FileSystem fs, Path configFilePath) {
-    if (!pathExists(fs, configFilePath)) {
+  /**
+   * Assuming that there's only 1 config.xml file in {@code jobFolderPath},
+   * this function parses config.xml and return a list of {@code JobConfig} objects.
+   * @param fs FileSystem object.
+   * @param jobFolderPath Path object of job directory.
+   * @return a list of {@code JobConfig} objects.
+   */
+  public static List<JobConfig> parseConfig(FileSystem fs, Path jobFolderPath) {
+    String[] jobFilesArr;
+    if (!pathExists(fs, jobFolderPath)) {
       return new ArrayList<>();
     }
+
+    try {
+      jobFilesArr = Arrays.stream(fs.listStatus(jobFolderPath))
+          .filter(f -> f.getPath().toString().endsWith("xml"))
+          .map(f -> f.getPath().toString())
+          .toArray(String[]::new);
+      Preconditions.checkArgument(jobFilesArr.length == 1);
+    } catch (IOException e) {
+      LOG.error("Failed to scan " + jobFolderPath.toString(), e);
+      return new ArrayList<>();
```

Comment:
I think it's more idiomatic to use `Collections.EMPTY_LIST`. Most implementations probably re-use a singleton empty list, which is more efficient than creating a new empty `ArrayList` each time this line of code is reached.

Reply:
👍

Reply:
Only thing I'd change is to use `Collections.emptyList()` instead so that it returns the correct `List` type.